tests/unit/test_loaders.py

```diff
@@ -36,46 +41,128 @@
 class TestJSONFileLoader(BaseEnvVar):
     def setUp(self):
         super().setUp()
-        self.data_path = os.path.join(os.path.dirname(__file__), 'data')
+        self.data_path = pathlib.Path(__file__).parent.joinpath('data')
         self.file_loader = JSONFileLoader()
-        self.valid_file_path = os.path.join(self.data_path, 'foo')
-        self.compressed_file_path = os.path.join(self.data_path, 'compressed')
+        self.valid_file_path = self.data_path.joinpath('foo')
+        self.compressed_file_path = self.data_path.joinpath('compressed')
+        if sys.version_info >= (3, 9):
+            self.zip_data_path = ZipPath(
+                self.data_path.joinpath('Archive.zip')
+            )
+            self.zip_valid_file_path = self.zip_data_path.joinpath('foo')
+            self.zip_compressed_file_path = self.zip_data_path.joinpath(
+                'compressed'
+            )
```

Comment:
Ideally we don't want to leak these details into the `setUp`. I think we'd talked about doing an explicit decorator for zip functionality. You can take a look at some examples [here](https://github.com/boto/botocore/blob/develop/tests/__init__.py#L56-L79).

We can either do this potentially as a pytest fixture or separate function. It doesn't look like the zip tests actually use anything from the class anyways, so the grouping may not be appropriate. We could consider moving these to their own test class.

Reply:
Ah ok makes sense! Yeah I was thinking some sort of decorator like `pytest.mark.skipif` would be useful here, but wasn't sure how it'd work without splitting everything into a separate class. This was definitely more of a lazy approach. I'll see what I can cobble together